### DISCUSSION TITLE
Is there StepFinishedEvent in allure2?

### DISCUSSION THREAD

**Discussion by mourav2222** [allure-framework/allure2#2065] [Questions & Support] [answered]
I want to save a screenshot each time at the end/finish of `@Step` and not add my saveScreenshotInAllure() function in each function with `@Step` annotation. I use a selenide framework yet.
How can i do this?

I do this so now:
```
   @Step("Logout...")
    public void logout() {

        sleep(2000);

        $("#dijit_form_DropDownButton_0").click();
        $("#dijit_form_Button_0_label").click();

      // Add in function
        saveScreenshotInAllure();
    }
```

**Comment by baev** [comment, marked as answer]
There is `StepLifecycleListener` extension point available. See https://stackoverflow.com/a/45693082 for details

**Comment by mourav2222** [comment]
Can i use this solution with AspectJ? But it doesn't work by me. 

![image](https://github.com/allure-framework/allure2/assets/76436807/73725e4e-89cc-4b81-bc84-b15d0b7c0e32)

I add a class StepAspect to src/java/test. But it doesn't execute the events, only in your Aspects calsses.

![image](https://github.com/allure-framework/allure2/assets/76436807/3a081214-59c5-466f-8ee3-9416395bf84d)

![image](https://github.com/allure-framework/allure2/assets/76436807/662756f3-2d0d-49d8-b379-ccb1e12d2cdb)

**Reply by baev** [reply]
But why? `StepLifecycleListener` will report all the steps, including the ones created by using `Allure#step` methods and via other modules such as AllureAssertJ.

**Reply by mourav2222** [reply]
Ok, i thougth it's better method. Ok i realized with StepLifecycleListener. thx.
